crates/block-producer/src/config.rs

```diff
@@ -42,9 +37,30 @@ impl Display for BlockProducerConfig {
 impl Default for BlockProducerConfig {
     fn default() -> Self {
         Self {
-            endpoint: Endpoint::localhost(DEFAULT_BLOCK_PRODUCER_PORT),
-            store_url: Endpoint::localhost(DEFAULT_STORE_PORT).to_string(),
+            endpoint: Url::parse(
+                format!("http://127.0.0.1:{DEFAULT_BLOCK_PRODUCER_PORT}").as_str(),
+            )
+            .unwrap(),
+            store_url: Url::parse(format!("http://127.0.0.1:{DEFAULT_STORE_PORT}").as_str())
+                .unwrap(),
```

Comment:
I thought it was weird that `Url` doesn't have a const constructor: https://github.com/servo/rust-url/issues/652. We could consider using the `url-macro` crate towards the bottom of the thread, or we could add it to our own set of macros since its rather trivial.

Though in the end I guess we get back to the issue that not all `Url` are bindable so we end up requiring the tests in any case.